code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -586,22 +585,19 @@ butcher
 		var/mob/living/critter/flock/drone/F = holder.owner
 		usr = F // don't ask, please, don't
 		if(F?.set_hand(1)) // grip tool
-			if(!F.is_in_hands(container_target)) //if it's not in any of our hands, pick it up
-				F.drop_item() // drop whatever we're holding
+			if(!F.is_in_hands(container_target))
+				F.drop_item()
 				F.set_dir(get_dir(F, container_target))
 				F.hand_attack(container_target) //try and pick it up
-			if(F.is_in_hands(container_target)) //did we do it?
+			if(F.is_in_hands(container_target))
 				if(F.absorber.item != container_target) //if it's in our manipulating hand
-					// we've picked up a container, just eat it, dump it onto the floor
 					container_target.MouseDrop(get_turf(F)) //this will do nothing on a locked secure storage
-					// might as well eat the container now
```

Comment:
useful comment

Reply:
I kind of merged this into the new comment